tutorial/examples/cytoscape/usage-phylogeny.py

```diff
@@ -0,0 +1,198 @@
+
+            support_id = clade_id + 's' + str(n)
+            child_id = clade_id + 'c' + str(n)
+            pos_y_child = row_positions[child] * ylen
+
+            cy_support_node = {
+                'data': {'id': support_id},
+                'position': {'x': pos_x, 'y': pos_y_child},
+                'grabbable': grabbable,
+                'classes': 'support'
+            }
+
+            cy_support_edge = {
+                'data': {
+                    'source': clade_id,
+                    'target': support_id,
+                    'sourceCladeId': clade_id
+                },
+            }
+
+            cy_edge = {
+                'data': {
+                    'source': support_id,
+                    'target': child_id,
+                    'length': clade.branch_length,
+                    'sourceCladeId': clade_id
+                },
+            }
+
+            if clade.confidence and clade.confidence.value:
+                cy_source['data']['confidence'] = clade.confidence.value
+
+            nodes.append(cy_support_node)
+            edges.extend([cy_support_edge, cy_edge])
+
+            add_to_elements(child, child_id)
+
+    col_positions = get_col_positions(tree)
+    row_positions = get_row_positions(tree)
+
+    nodes = []
+    edges = []
+
+    add_to_elements(tree.clade, 'r')
+
+    return nodes, edges
+
+# path = os.path.abspath('data/apaf.xml')
+# Define elements, stylesheet and layout
+tree = Phylo.read('tutorial/examples/cytoscape/data/apaf.xml', 'phyloxml')
+nodes, edges = generate_elements(tree)
+elements = nodes + edges
+
+layout = {'name': 'preset'}
+
+stylesheet = [
+    {
+        'selector': '.nonterminal',
+        'style': {
+            'label': 'data(confidence)',
+            'background-opacity': 0,
+            "text-halign": "left",
+            "text-valign": "top",
+        }
+    },
+    {
+        'selector': '.support',
+        'style': {'background-opacity': 0}
+    },
+    {
+        'selector': 'edge',
+        'style': {
+            "source-endpoint": "inside-to-node",
+            "target-endpoint": "inside-to-node",
+        }
+    },
+    {
+        'selector': '.terminal',
+        'style': {
+            'label': 'data(name)',
+            'width': 10,
+            'height': 10,
+            "text-valign": "center",
+            "text-halign": "right",
+            'background-color': '#222222'
+        }
+    }
+]
+
+# Start the app
+app = dash.Dash(__name__)
+
+app.layout = html.Div([
+    dash_cytoscape.Cytoscape(
+        id='cytoscape-usage-phylogeny',
+        elements=elements,
+        stylesheet=stylesheet,
+        layout=layout,
+        style={
+            'height': '95vh',
+            'width': '100%'
+        }
+    )
+])
+
+
+@app.callback(Output('cytoscape-usage-phylogeny', 'stylesheet'),
+              [Input('cytoscape-usage-phylogeny', 'mouseoverEdgeData')])
+def color_children(edgeData):
+    if not edgeData:
```

Comment:
```suggestion
    if edgeData is not None:
```
Actually, is it what you mean? There might be some type conversion going on...
Reference: https://www.python.org/dev/peps/pep-0008/#programming-recommendations

Reply:
Will update in next commit